GeneratorInterface/Core/interface/ConcurrentGeneratorFilter.h

```diff
@@ -407,6 +449,9 @@ namespace edm {
     if (rCache->product_.compare_exchange_strong(expect, griproduct.get())) {
       griproduct.release();
     }
+    if (cache == useInLumi_.load()) {
```

Comment:
If `useInLumi_` is set to nullptr in `globalBeginLuminosityBlockProduce` will this ever happen?

Reply:
Yes. streamEndLuminosityBlockSummary selects the stream cache to use in globalBeginLumi and that occurs before streamEndRun. So it will happen on the stream with the selected stream cache.